XenAdmin/ConsoleView/RdpClient.cs

```diff
@@ -123,15 +128,28 @@ public Point rdpLocationOffset
             }
         }
 
-        private void RDPAddOnDisconnected()
+        private void AddRDPEventHandlers()
         {
             if (rdpControl == null)
                 return;
 
-            if (rdpClient9 == null)
-                rdpClient6.OnDisconnected += rdpClient_OnDisconnected;
-            else
-                rdpClient9.OnDisconnected += rdpClient_OnDisconnected;
+            var rdpClient = (IRdpClient)rdpClient9 ?? rdpClient6;
+            if (rdpClient == null)
+            {
+                return;
+            }
+
+            rdpClient.OnDisconnected += (_, e) =>
+            {
+                Program.AssertOnEventThread();

```

Comment:
Not your change, but why is an assertion needed here? Is it likely that it will be called from a thread other than the UI? Alternatively, if that were likely, then you'd probably want to make _connecting and _authWarningvisible volatile.

Reply:
> Not your change, but why is an assertion needed here? Is it likely that it will be called from a thread other than the UI? Alternatively, if that were likely, then you'd probably want to make _connecting and _authWarningvisible volatile (because that might be the case for the rest of the events).

I actually remember thinking the same. Wasn't going to mess with it though 😂 

I'd leave it since in the future we'd like to refactor this? 🤞